devtools/pip-tools/pr_tool/backport.py

```diff
@@ -0,0 +1,317 @@
+#   BAREOS® - Backup Archiving REcovery Open Sourced
+#
+#   Copyright (C) 2023-2023 Bareos GmbH & Co. KG
+#
+#   This program is Free Software; you can redistribute it and/or
+#   modify it under the terms of version three of the GNU Affero General Public
+#   License as published by the Free Software Foundation and included
+#   in the file LICENSE.
+#
+#   This program is distributed in the hope that it will be useful, but
+#   WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+#   Affero General Public License for more details.
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program; if not, write to the Free Software
+#   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+#   02110-1301, USA.
+
+"""
+this module provides commands to do a backport of a specific
+bareos PR into another branch
+"""
+
+import logging
+from os import environ
+from sys import stdout, stderr
+from subprocess import run
+from tempfile import NamedTemporaryFile
+from importlib import resources
+from git import RemoteReference
+from git.exc import GitCommandError
+from .github import Gh, InvokationError, whoami
+
+
+class UnmanagedBranchException(Exception):
+    pass
+
+
+def git_editor(repo, file):
+    if "GIT_EDITOR" in environ:
+        editor = environ["GIT_EDITOR"]
+    else:
+        editor = repo.git.var("GIT_EDITOR")
+    res = run([editor, file], check=False)
+    return res.returncode == 0
+
+
+def find_user_remote(repo, username):
+    expected_url = f"git@github.com:{username}/bareos.git"
+    for remote in repo.remotes:
+        if expected_url == list(remote.urls)[0]:
+            return remote
+    return None
+
+
+def get_branch_by_name(repo, name):
+    for branch in repo.branches:
+        if branch.name == name:
+            return branch
+    return None
+
+
+def resolve_target_branch(repo, git_remote, branch_name):
+    if branch_name is None:
+        branch_name = repo.active_branch.name
+    local_branch = get_branch_by_name(repo, branch_name)
```

Comment:
The problem with user `repo.active_branch` instead of `repo.active_branch.name` is that the types wouldn't match anymore.
We either get `branch_name` passed in as a string (as it is coming from the command line argument `--into`) and `repo.active_branch.name` is the fallback to the current branch.

Using a branch object in one case and a string in the other doesn't seem like good practice to me.

However, I totally agree that it is probably a good idea to improve error handling here.

Reply:
But local_branch is always a branch is it not ?